k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/CredentialsResourceHandler.kt

```diff
@@ -0,0 +1,102 @@
+package com.amazon.spinnaker.keel.k8s.resolver
+
+import com.amazon.spinnaker.keel.k8s.CREDENTIALS_RESOURCE_SPEC_V1
+import com.amazon.spinnaker.keel.k8s.K8sData
+import com.amazon.spinnaker.keel.k8s.K8sObjectManifest
+import com.amazon.spinnaker.keel.k8s.model.CredentialsResourceSpec
+import com.amazon.spinnaker.keel.k8s.service.CloudDriverK8sService
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
+import com.netflix.spinnaker.keel.api.plugins.Resolver
+import com.netflix.spinnaker.keel.api.support.EventPublisher
+import com.netflix.spinnaker.keel.orca.OrcaService
+import retrofit2.HttpException
+import java.util.*
+
+class CredentialsResourceHandler(
+    cloudDriverK8sService: CloudDriverK8sService, taskLauncher: TaskLauncher, eventPublisher: EventPublisher,
+    orcaService: OrcaService, resolvers: List<Resolver<*>>
+) : GenericK8sResourceHandler<CredentialsResourceSpec, K8sObjectManifest>(
+    cloudDriverK8sService, taskLauncher,
+    eventPublisher, orcaService, resolvers
+) {
+    override val supportedKind = CREDENTIALS_RESOURCE_SPEC_V1
+    private val encoder: Base64.Encoder = Base64.getMimeEncoder()
+
+    public override suspend fun toResolvedType(resource: Resource<CredentialsResourceSpec>): K8sObjectManifest {
+        val cred = cloudDriverK8sService.getCredentialsDetails(resource.serviceAccount, resource.spec.template.data?.account as String)
```

Comment:
we need to do better handling of errors here. right now, if a 404 or 400 is returned, this call returns a `null` cred which results in resource resolution to break later in the code.

Reply:
This now catches `HttpException` and throws `ResourceCurrentlyUnresolvable`. Unit tests have been added for them as well .